muxers/yamux/src/lib.rs

```diff
@@ -290,6 +290,13 @@ impl YamuxConfig {
         self
     }
 
+    /// Set the max. payload size used when sending data frames. Payloads larger
+    /// than the configured max. will be split.
+    pub fn set_split_send_size(&mut self, n: usize) -> &mut Self {
```

Comment:
Out of curiosity, why do you need this flag?

Reply:
We had a hard time debugging issue when we encountered UnexpectedEOF errors. Payload data was `N * split_send_size`, so we thought that it can affect the problem, and wondered why it is impossible to change this value. The reason was that we forgot to close stream after write and it got dropped immediately after write. 

Regarding this issue, i thought that it can be useful to add panic, assert or at least warning when garbage collecting stream in "Open" state [here](https://github.com/libp2p/rust-yamux/blob/058010157e5c58976ce0fe78a84bf88ec9dd8656/src/connection.rs#L1003). I have seen that you claim that `poll_close` must be called before dropping the stream [here](https://github.com/libp2p/rust-yamux/issues/117#issuecomment-824027292), so it may help users catch the same issue before it leads to serious consequences.

Reply:
Also you can add a comment in the examples mentioning the importance of the closing of the socked, like [here](https://github.com/libp2p/rust-libp2p/blob/master/examples/file-sharing.rs#L711). I can make a PR or issue if you agree

Reply:
> Also you can add a comment in the examples mentioning the importance of the closing of the socked, like [here](https://github.com/libp2p/rust-libp2p/blob/master/examples/file-sharing.rs#L711). I can make a PR or issue if you agree

Sure. That sounds good to me @ValeryAntopol.